perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -343,55 +345,58 @@ public boolean isGroupAssigned(PerunSession sess, Group group, Resource resource
 
 
 	@Override
-	public void assignGroupToResource(PerunSession sess, Group group, Resource resource) throws WrongAttributeValueException, WrongReferenceAttributeValueException, GroupAlreadyAssignedException {
-		Vo groupVo = getPerunBl().getGroupsManagerBl().getVo(sess, group);
+	public void assignGroupToResource(PerunSession sess, Group group, Resource resource) throws WrongAttributeValueException, WrongReferenceAttributeValueException, GroupAlreadyAssignedException, GroupResourceMismatchException {
+		assignGroupsToResource(sess, Collections.singletonList(group), resource);
+	}
 
-		// Check if the group and resource belongs to the same VO
-		if (!groupVo.equals(this.getVo(sess, resource))) {
-			throw new InternalErrorException("Group " + group + " and resource " + resource + " belongs to the different VOs");
+	@Override
+	public void assignGroupsToResource(PerunSession perunSession, List<Group> groups, Resource resource) throws WrongAttributeValueException, WrongReferenceAttributeValueException, GroupAlreadyAssignedException, GroupResourceMismatchException {
+		Set<Member> members = new HashSet<>();
+		boolean processAttributes = true;
+		// if there are no assigned services, no attributes have to be checked or filled
+		if (getAssignedServices(perunSession, resource).isEmpty()) {
```

Comment:
I think that since the members.addAll phase was skipped in group cycle, the fill/set/check for user/member attributes was skipped as well, but it was quite messy. I hope it's nicer and clearer now.

Reply:
Ok, I probably overlooked it in the diff. Thank you. Its clear now.